contracts/XVSVault/XVSVault.sol

```diff
@@ -0,0 +1,359 @@
+            uint256 reward =
+                multiplier.mul(rewardTokenPerBlock).mul(pool.allocPoint).div(
+                    totalAllocPoints[_rewardToken]
+                );
+            accRewardPerShare = accRewardPerShare.add(
+                reward.mul(1e12).div(supply)
+            );
+        }
+        return user.amount.mul(accRewardPerShare).div(1e12).sub(user.rewardDebt);
+    }
+
+    // Update reward vairables for all pools. Be careful of gas spending!
+    function massUpdatePools(address _rewardToken) public {
+        uint256 length = poolInfos[_rewardToken].length;
+        for (uint256 pid = 0; pid < length; ++pid) {
+            updatePool(_rewardToken, pid);
+        }
+    }
+
+    // Update reward variables of the given pool to be up-to-date.
+    function updatePool(address _rewardToken, uint256 _pid) public {
+        PoolInfo storage pool = poolInfos[_rewardToken][_pid];
+        if (block.number <= pool.lastRewardBlock) {
+            return;
+        }
+        uint256 supply = pool.token.balanceOf(address(this));
+        if (supply == 0) {
+            pool.lastRewardBlock = block.number;
+            return;
+        }
+        uint256 curBlockNumber = block.number;
+        uint256 multiplier = curBlockNumber.sub(pool.lastRewardBlock);
+        uint256 reward =
+            multiplier.mul(rewardTokenAmountsPerBlock[_rewardToken]).mul(pool.allocPoint).div(
+                totalAllocPoints[_rewardToken]
+            );
+        pool.accRewardPerShare = pool.accRewardPerShare.add(
+            reward.mul(1e12).div(supply)
+        );
+        pool.lastRewardBlock = block.number;
+    }
+
+    // Get user info with reward token address and pid
+    function getUserInfo(
+        address _rewardToken,
+        uint256 _pid,
+        address _user
+    )
+    public view returns (uint256 amount, uint256 rewardDebt) {
+        UserInfo storage user = userInfos[_rewardToken][_pid][_user];
+        amount = user.amount;
+        rewardDebt = user.rewardDebt;
+    }
+
+    /**
+     * @notice Determine the xvs stake balance for an account
+     * @param account The address of the account to check
+     * @param blockNumber The block number to get the vote balance at
+     * @return The balance that user staked
+     */
+    function getPriorVotes(address account, uint256 blockNumber) external view returns (uint96) {
+        require(blockNumber < block.number, "XVSVault::getPriorVotes: not yet determined");
+        require(xvsAddress != address(0), "XVSVault:getPriorVotes: xvs address is not set");
+
+        PoolInfo[] storage poolInfo = poolInfos[xvsAddress];
+
+        uint256 length = poolInfo.length;
+        for (uint256 pid = 0; pid < length; ++pid) {
+            if (address(poolInfo[pid].token) == address(xvsAddress)) {
+                UserInfo storage user = userInfos[xvsAddress][pid][account];
+                return uint96(user.amount);
+            }
+        }
+        return uint96(0);
+    }
```

Comment:
This function just returns the user's XVS balance in the vault, not the voting power, is this supposed to be changed in the governance PR?